instance/tests/test_instance_upgrade.py

```diff
@@ -0,0 +1,252 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2016 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance Upgrade - Tests
+"""
+from unittest.mock import call, patch, Mock
+import ddt
+
+from instance.instance_upgrade import InstanceUpgrade, DogwoodToEucalyptus1, GitVersionSpec, Eucalyptus1toEucalyptus2
+from instance.models.openedx_instance import OpenEdXInstance
+from instance.tests.base import TestCase
+
+
+class BaseInstanceUpgradeTest(TestCase):
+    """
+    Base class for instance upgrade tests
+    """
+    def setUp(self):
+        """
+        Test setup
+        """
+        super(BaseInstanceUpgradeTest, self).setUp()
+        self.instance_objects_mock = self._apply_patch(OpenEdXInstance, patch_attribute='objects')
+        self.spawn_appserver_mock = self._apply_patch("instance.instance_upgrade.spawn_appserver")
+
+    def _apply_patch(self, target, new=None, patch_attribute=False, **kwargs):
```

Comment:
@e-kolpakov Setting `patch_attribute` to `None` instead of `False` by default would be more appropriate I think (since you're using it to specify names of attributes to patch).

Also, you might be able to remove the `new` parameter. AFAICT, none of the calls to `_apply_path` actually specify it.